SpatialGDK/Source/SpatialGDK/Public/Utils/SchemaUtils.h

```diff
@@ -214,6 +214,46 @@ inline FVector GetVectorFromSchema(Schema_Object* Object, Schema_FieldId Id)
 // Does not clear OutPath first.
 void GetFullPathFromUnrealObjectReference(const FUnrealObjectRef& ObjectRef, FString& OutPath);
 
+template <typename T>
+struct TSpatialComponent
```

Comment:
Is this meant to be a shim to keep the old component types relevant?

Reply:
We seem to be using it for static polymorphism but I can't see why we need to have this

Reply:
Ok have read through it now. This seems like a lot of indirection to get a function at the end which will process a component update and do whatever with it. Why not just do this at at the view delta or when you want to write stuff and saves several function jumps that make it a bit hard to work out what's going on?

e.g. for reading, when iterating over some sub view delta, get the updates for manifests and then apply them to whatever that systems data store is.

I guess it functionally isn't that difference but I think the code you get will be much simpler.

Reply:
Or to put it another way - I think abstraction is adding more mental overhead than it is reducing.

Reply:
The main reason I've written this is to cut new C++ components down from:
```
struct FMyComponent
{
	static constexpr Worker_ComponentId ComponentId = 1337;

	FMyComponent(const ComponentData& Data)
	{
		ApplyComponentData(Data);
	}

	ComponentData CreateComponentData() const
	{
		ComponentData Data(ComponentId);
		WriteSchema(*Data.GetFields());
		return MoveTemp(Data);
	}

	ComponentUpdate CreateComponentUpdate() const
	{
		ComponentUpdate Update(ComponentId);
		WriteSchema(*Update.GetFields());
		return MoveTemp(Update);
	}

	void ApplyComponentData(const ComponentData& Data)
	{
		ApplySchema(*Data.GetFields());
	}

	void ApplyComponentUpdate(const ComponentUpdate& Update)
	{
		ApplySchema(*Update.GetFields());
	}

	void ApplySchema(const Schema_Object& Object)
	{
		// Actually important code 
	}

	void WriteSchema(Schema_Object& Object)
	{
		// Actually important code
	}
}
```
to
```
struct FMyComponent : public TSpatialComponent<FMyComponent>
{
	static constexpr Worker_ComponentId ComponentId = 1337;

	FMyComponent(const ComponentData& Data)
	{
		ApplyComponentData(Data);
	}

	void ApplySchema(const Schema_Object& Object)
	{
		// Actually important code 
	}

	void WriteSchema(Schema_Object& Object)
	{
		// Actually important code
	}
}
```
I'm not sure I'm following the view delta reasoning but one could use this just as they'd use any other C++ component, just with less lines of boilerplate code.

Reply:
I did remove this as you've suggested though :D This PR is probably not the place to introduce templates such as this one.

Reply:
I think the bigger question I have is why do we need all that component stuff in the first place?

Reply:
As in, why have the C++ bindings? They are convenient, especially as we use a `TSet` in C++ to hasten comparisons, and `CreateComponentData`/`CreateComponentUpdate` is a fast an easy way to create one from the other.
I could definitely change `CreateComponentUpdate` into a special method as it's only ever used on the populating worker - so it can be limited to the `populated_entities` set, limiting egress.